backend/src/ledgers.ts

```diff
@@ -75,42 +148,57 @@ export async function updateLedgers() {
     .limit(200)
     .stream({
       onmessage: async (ledger: LedgerRecord) => {
-        await updateCache([ledger], REDIS_LEDGER_KEY, REDIS_PAGING_TOKEN_KEY);
+        for (const interval of intervalTypes) {
+          await updateCache(
+            [ledger],
+            REDIS_LEDGER_KEYS[interval],
+            REDIS_PAGING_TOKEN_KEY[interval],
+          );
+        }
       },
     });
 }
 
 export async function catchup(
   ledgersKey: string,
-  pagingTokenStart: string,
+  pagingToken: string,
   pagingTokenKey: string,
-  limit: number, // if 0, catchup until now
+  limit: number,
+  interval: INTERVALS = INTERVALS.month,
+  total: number = 0,
 ) {
   const horizon = new stellarSdk.Server("https://horizon.stellar.org");
+  const resp = await horizon
+    .ledgers()
+    .cursor(pagingToken)
+    .limit(HORIZON_LIMITS[interval])
+    .call();
   let ledgers: LedgerRecord[] = [];
-  let total = 0;
-  let pagingToken = pagingTokenStart;
-
-  // eslint-disable-next-line no-constant-condition
-  while (true) {
-    // eslint-disable-next-line no-await-in-loop
-    const resp = await horizon.ledgers().cursor(pagingToken).limit(200).call();
-    ledgers = resp.records;
-    total += resp.records.length;
-    if (ledgers.length === 0 || (limit && total > limit)) {
-      break;
-    }
 
-    pagingToken = ledgers[ledgers.length - 1].paging_token;
-    // eslint-disable-next-line no-await-in-loop
-    await updateCache(ledgers, ledgersKey, pagingTokenKey);
+  ledgers = resp.records;
+  total += resp.records.length;
+
+  if (ledgers.length === 0 || (limit && total > limit)) {
+    return;
   }
+  pagingToken = ledgers[ledgers.length - 1].paging_token;
+  await updateCache(ledgers, ledgersKey, pagingTokenKey, interval);
+
+  await catchup(
+    ledgersKey,
+    pagingToken,
+    pagingTokenKey,
+    limit,
+    interval,
+    total,
+  );
 }
 
 export async function updateCache(
   ledgers: LedgerRecord[],
   ledgersKey: string,
   pagingTokenKey: string,
+  interval = INTERVALS.month,
```

Comment:
This is causing an error looks like when streaming new ledgers. I suggest we remove the default values for interval, to make sure the right one is getting passed in
```suggestion
  interval: INTERVAL,
```